lib/mongoid/contextual/mongo.rb

```diff
@@ -161,9 +161,16 @@ def each(&block)
       # @note We don't use count here since Mongo does not use counted
       #   b-tree indexes.
       #
+      # @param [ Hash | BSON::ObjectId | String ] id_or_conditions an _id to
+      #   search for, or a hash of conditions.
+      #
       # @return [ true | false ] If the count is more than zero.
-      def exists?
-        !!(view.projection(_id: 1).limit(1).first)
+      def exists?(id_or_conditions = nil)
+        case id_or_conditions
+        when nil then !!(view.projection(_id: 1).limit(1).first)
+        when Hash then Mongo.new(criteria.where(id_or_conditions)).exists?
+        else Mongo.new(criteria.where(_id: id_or_conditions)).exists?
```

Comment:
I don't love explicitly creating a new Contextual object, but this is the only way to verify that a `Contextual::Mongo` object is used for the later `exists?` and not a `Memory`... 

I believe there can be a situation where calling exists? on a criteria in this method or the Memory method will result in the opposite contextual object being created and used. Explicitly calling the constructors here, instead of allowing create_context to do it, prevents that.